rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2070,11 +2071,15 @@ static rmw_ret_t publish_loaned_int(
 
   // if the publisher allow loaning
   if (cdds_publisher->is_loaning_available) {
-    TRACETOOLS_TRACEPOINT(rmw_publish, ros_message);
+
     auto d = new serdata_rmw(cdds_publisher->sertype, ddsi_serdata_kind::SDK_DATA);
     d->iox_chunk = ros_message;
     // since we write the loaned chunk here, set the data state to raw
     shm_set_data_state(d->iox_chunk, IOX_CHUNK_CONTAINS_RAW_DATA);
+    // the following wouldn't work since dds_writecdr resets the timestamp anyways.
+    // const dds_time_t tstamp = dds_time();
+    // d->timestamp.v = tstamp;
+    TRACETOOLS_TRACEPOINT(rmw_publish, (const void *)publisher, ros_message, 0);
```

Comment:
This is a bit unfortunate. As of now `dds_writecdr` will always overwrite the timestamp.

Will the serdata object outlive the `dds_writecdr` function call? If so, we could maybe also take the timestamp from it after the fact.

Reply:
Ok, I had a closer look into cyclonedds and it looks like the serdata object does _not_ outlive `dds_writecdr` if everything is fine.
@christophebedard I'd suggest to leave it like this for the time beeing. The only way I could image that this could be fixed is add the respective API to cyclonedds.

Reply:
After having a closer look at cyclones API, it turns out that there is an APi which seems to fit this usecase quite well: `dds_forwardcdr`. @eboasson and/or @ivanpauno, it would be great if you could have a look at these changes (again).

Reply:
@cwecht I'm catching up with whatever happened during my break ... I'll review as request but wanted to quickly note that this weird behaviour of `dds_writecdr` is the reason I needed to fix something ... and added `dds_forwardcdr`.